guvnor-ala/guvnor-ala-services-rest/src/test/java/org/guvnor/ala/services/rest/tests/RestPipelineEventsTest.java

```diff
@@ -0,0 +1,210 @@
+        deployment.addClass( FunctionConfigExecutor.class );
+        deployment.addClass( ConfigExecutor.class );
+        deployment.addClass( ProviderConfig.class );
+        deployment.addClass( Provider.class );
+        deployment.addClass( DockerRuntimeConfig.class );
+        deployment.addClass( DockerRuntime.class );
+        deployment.addClass( RuntimeBuilder.class );
+        deployment.addClass( DockerRuntimeExecExecutor.class );
+        deployment.addClass( RuntimeDestroyer.class );
+        deployment.addClass( DockerAccessInterface.class );
+        deployment.addClass( DockerAccessInterfaceImpl.class );
+        deployment.addClass( Disposable.class );
+        deployment.addClass( DockerException.class );
+        deployment.addClass( DockerRuntimeManager.class );
+        deployment.addClass( RuntimeManager.class );
+        deployment.addClass( org.guvnor.ala.config.Config.class );
+        deployment.addClass( org.guvnor.ala.config.ProviderConfig.class );
+        deployment.addClass( org.guvnor.ala.docker.config.DockerProviderConfig.class );
+        deployment.addClass( org.guvnor.ala.docker.model.DockerProvider.class );
+        deployment.addClass( org.guvnor.ala.pipeline.FunctionConfigExecutor.class );
+        deployment.addClass( BiFunctionConfigExecutor.class );
+        deployment.addClass( org.guvnor.ala.registry.RuntimeRegistry.class );
+        deployment.addClass( org.guvnor.ala.runtime.providers.ProviderBuilder.class );
+        deployment.addClass( org.guvnor.ala.runtime.providers.ProviderDestroyer.class );
+        deployment.addClass( org.guvnor.ala.runtime.providers.ProviderId.class );
+
+        deployment.addClass( GitConfigExecutor.class );
+        deployment.addClass( MavenProjectConfig.class );
+        deployment.addClass( MavenProjectConfigExecutor.class );
+        deployment.addClass( Project.class );
+        deployment.addClass( Source.class );
+        deployment.addClass( ProjectConfig.class );
+        deployment.addClass( BuildConfig.class );
+        deployment.addClass( MavenBuildConfig.class );
+        deployment.addClass( MavenBuildConfigExecutor.class );
+        deployment.addClass( MavenBuildExecConfigExecutor.class );
+        deployment.addClass( DockerBuildConfigExecutor.class );
+        deployment.addClass( DockerProviderConfigExecutor.class );
+        deployment.addClass( DockerProvisioningConfigExecutor.class );
+
+        deployment.addClass( MockPipelineEventListener.class );
+
+        deployment.addAsManifestResource( EmptyAsset.INSTANCE, "beans.xml" );
+        return deployment;
+    }
+
+    @Before
+    public void setUp() {
+        try {
+            tempPath = Files.createTempDirectory( "zzz" ).toFile();
+        } catch ( IOException e ) {
+            e.printStackTrace();
+        }
+    }
+
+    @After
+    public void tearDown() {
+        FileUtils.deleteQuietly( tempPath );
+    }
+
+    @Test
+    public void testEventsPropagation() {
```

Comment:
@mbarkley 
I kinda agree on this one. I would rather use pure JUnit tests with Mocks, but I would also keep this test as it is with its counterpart plain JUnit test. Mostly because I would like to make sure that if we change the scope of the beans in any place along the pipeline components this test and the other Arquillian Tests breaks. Wdyt?